src/hooks/useInfiniteQueryWithMeta.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  QueryFunction,
+  QueryKey,
+  UseInfiniteQueryOptions,
+  useInfiniteQuery,
+} from "@tanstack/react-query";
+import type { WithPageMeta } from "@/interfaces/article";
+
+export const useInfiniteQueryWithMeta = <TData extends WithPageMeta<{}>>(
+  queryKey: QueryKey,
+  fetchFn: QueryFunction<TData>,
+  config?: UseInfiniteQueryOptions<TData>
+) => {
```

Comment:
`<{}>` 안하려면 이런방법도 있을거같은데 준서님 편하신대로 가시죠

```suggestion
export const useInfiniteQueryWithMeta = <T>(
  queryKey: QueryKey,
  fetchFn: QueryFunction<WithPageMeta<T>>,
  config?: UseInfiniteQueryOptions<WithPageMeta<T>>
) => {
```